### ISSUE TITLE
[bug] fatal: bad object upon upgrade to 1.30.0

### ISSUE THREAD

**Issue by CluEleSsUK** [moonrepo/moon#1734]
**Describe the bug**

upon running `moon ci` the following error occurs:

```
▪▪▪▪ installing proto 0.42.2
▪▪▪▪ Gathering touched files
Error: process::failed

  × Process git failed: exit code 128 
  │ 
  │ fatal: bad object c2ae0b5c684cdd3f689aa60969866e25f81ab645
  
```

where `c2ae0b5c684cdd3f689aa60969866e25f81ab645` is the commit hash of the head of the master branch

**Environment**

```
  System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Max
    Memory: 856.08 MB / 32.00 GB
    Shell: 3.7.1 - /usr/local/bin/fish
  Binaries:
    Node: 22.3.0 - ~/.nvm/versions/node/v22.3.0/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v22.3.0/bin/yarn
    npm: 10.9.0 - ~/.nvm/versions/node/v22.3.0/bin/npm
  Managers:
    Cargo: 1.79.0 - ~/.cargo/bin/cargo
    Homebrew: 4.4.6 - /opt/homebrew/bin/brew
    pip3: 24.2 - /opt/homebrew/bin/pip3
    RubyGems: 3.0.3.1 - /usr/bin/gem

```

**Comment by CluEleSsUK** [comment]
Not running a shallow clone or anything fwiw - happens on both github runners and locally

**Comment by CluEleSsUK** [comment]
some more context: the `curl | bash` version worked fine inside a docker image running linux, so I tried `curl | bash`ing the latest version locally, and now it works for my locally - it seems that having `1.29.x` in the package.json and `1.30.x` locally or the other way round is causing the problems. This is somewhat unexpected for a minor version upgrade though :(

**Comment by milesj** [comment]
@CluEleSsUK What's the `--log trace` output for that git error? I need to know what git command is actually causing this.

**Comment by milesj** [comment]
That error kind of seems like the repo is in a bad state? https://www.swyx.io/solve-git-bad-object-head

**Comment by harlequin** [comment]
@CluEleSsUK Are you running in Azure DevOps or any other system?

**Comment by milesj** [comment]
@harlequin Thanks for this, that's super helpful. Should be an easy fix!

**Comment by CluEleSsUK** [comment]
sorry for late response - I was getting this both locally and on our self-hosted github runner in GCP.
We do have a bunch of submodules though, so it could be related.

**Comment by milesj** [comment]
Should be fixed in 1.30.1
